src/mixer/basetrackplayer.h

```diff
@@ -100,6 +107,10 @@ class BaseTrackPlayerImpl : public BaseTrackPlayer {
     ControlObjectSlave* m_pPreGain;
     ControlObjectSlave* m_pSpeed;
     ControlObjectSlave* m_pPitchAdjust;
+    QScopedPointer<ControlObjectSlave> m_pInputEnabled;
+    QScopedPointer<ControlObjectSlave> m_pPassthroughEnabled;
+    QScopedPointer<ControlObjectSlave> m_pVinylControlEnabled;
+    QScopedPointer<ControlObjectSlave> m_pVinylControlStatus;
```

Comment:
Are you using -O3 and a release build (to drop the Q_ASSERTs? I did and the
assembly was identical.

On Wed, Jan 20, 2016, 3:24 AM Daniel Schürmann <notifications@github.com>
wrote:

> In src/mixer/basetrackplayer.h
> <https://github.com/mixxxdj/mixxx/pull/858#discussion_r50242825>:
>
> > @@ -100,6 +107,10 @@ class BaseTrackPlayerImpl : public BaseTrackPlayer {
> >      ControlObjectSlave* m_pPreGain;
> >      ControlObjectSlave* m_pSpeed;
> >      ControlObjectSlave* m_pPitchAdjust;
> > +    QScopedPointer<ControlObjectSlave> m_pInputEnabled;
> > +    QScopedPointer<ControlObjectSlave> m_pPassthroughEnabled;
> > +    QScopedPointer<ControlObjectSlave> m_pVinylControlEnabled;
> > +    QScopedPointer<ControlObjectSlave> m_pVinylControlStatus;
>
> .. and these are working how we might expect it:
> boost::scoped_ptr and the deprecated std::auto_ptr
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mixxxdj/mixxx/pull/858/files#r50242825>.
>